packages/ove-asset-manager/src/OVE.Service.AssetManager/Domain/IAssetFileOperations.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using OVE.Service.AssetManager.Models;
+
+namespace OVE.Service.AssetManager.Domain {
+    public interface IAssetFileOperations {
+        string ResolveFileUrl(OVEAssetModel asset);
+        // ReSharper disable twice UnusedParameter.Global
+        Task<bool> MoveFile(OVEAssetModel oldAsset, OVEAssetModel newAsset);
+        Task<bool> DeleteFile(OVEAssetModel oveAssetModel);

```

Comment:
I am sniffing - there seems to be smelly code here, :).

I think OVEAssetModel can/should have a better name. Why don't we call it OVEAsset? Why is the word Model required? Also, why are these methods ending with File? Is there a scenario where you delete a file and not the asset? It is slightly confusing to understand what's happening here.

Reply:
Following standard practice in asp.net is to append Model and Controller to such classes. 
I can break this practice if preferred? 

This class deals with files. The controller class deals with database. 
The controller class uses this class to deal with files in tandem. 

The controller class os decoupled from where the files are stored or interacting with them via this interface.